### PR TITLE
Use stream api in swift package

### PR DESCRIPTION
**Pull request by nguyentruongtho**
Let's do this first.

### CODE REVIEW

File: src/com/facebook/buck/swift/SwiftLibrary.java [hunk 1]
```diff
@@ -128,12 +131,16 @@ private boolean isPlatformSupported(CxxPlatform cxxPlatform) {
     if (!isPlatformSupported(cxxPlatform)) {
       return ImmutableList.of();
     }
-    return FluentIterable.from(exportedDeps)
-        .filter(NativeLinkable.class)
-        .append(
-            new SwiftRuntimeNativeLinkable(
-                swiftPlatformFlavorDomain.getValue(
-                    cxxPlatform.getFlavor())));
+    SwiftRuntimeNativeLinkable swiftRuntimeNativeLinkable = new SwiftRuntimeNativeLinkable(
```
Comment (illicitonion) [inline src/com/facebook/buck/swift/SwiftLibrary.java]: This one *would* be cleaner as:

```
RichStream.from(exportedDeps)
  .filter(NativeLinkable.class)
  .concat(RichStream.of(new SwiftRuntimeNativeLinkable...))
  .collect(MoreCollectors.toImmutableSet());
```

File: src/com/facebook/buck/swift/SwiftLibraryDescription.java [hunk 2]
```diff
@@ -311,9 +312,13 @@ private BuildRule createSharedLibraryBuildRule(
         Optional.of(sharedLibrarySoname),
         sharedLibOutput,
         Linker.LinkableDepType.SHARED,
-        FluentIterable.from(params.getDeps())
-            .filter(NativeLinkable.class)
-            .append(swiftRuntimeLinkable),
+        ImmutableSet.<NativeLinkable>builder()
```
Comment (illicitonion) [inline src/com/facebook/buck/swift/SwiftLibraryDescription.java]: RichStream.concat

**Comment by nguyentruongtho** [comment]
RichStream is nice, I feel very much outdated... Will update the PR soon.

**Comment by aiked** [comment]
@nguyentruongtho thank you for this quick fix!

**Pull request closed without merging**